packages/grid/_modules_/grid/hooks/features/columnMenu/useGridColumnMenu.ts

```diff
@@ -76,5 +54,6 @@ export const useGridColumnMenu = (apiRef: GridApiRef): void => {
     'ColumnMenuApi',
   );
 
-  useGridApiEventHandler(apiRef, GridEvents.columnResizeStart, handleColumnResizeStart);
+  useGridApiEventHandler(apiRef, GridEvents.columnResizeStart, hideColumnMenu);
```

Comment:
`hideColumnMenu` is also resetting `state.columnMenu.field`, is it changing the behavior ?

```tsx
// handleColumnResizeStart
columnMenu: { ...state.columnMenu,  open: false }

// hideColumnMenu
columnMenu: { ...state.columnMenu, open: false, field: undefined },
```

Reply:
There's no problem in not resetting the field. When the menu is open it always changes the field:

https://github.com/mui-org/material-ui-x/blob/8a8828dca0c9c0c1264b955ab76a84b566fd54a6/packages/grid/_modules_/grid/hooks/features/columnMenu/useGridColumnMenu.ts#L18